test/cpp/qps/qps_worker.cc

```diff
@@ -199,6 +199,16 @@ class WorkerServiceImpl final : public WorkerService::Service {
     if (!client) {
       return Status(StatusCode::INVALID_ARGUMENT, "Couldn't create client");
     }
+    if (!client->AllChannelsConnected()) {
+      if (worker_->ShouldDieOnConnectionFailure()) {
+        grpc_core::Crash("Client failed to connect to all channels");
+      } else {
+        client->AwaitThreadsCompletion();
+        ctx->TryCancel();
```

Comment:
I don't think `TryCancel` is necessary?

Simply terminating the RPC with the status code and message below seems ideal

Reply:
`TryCancel` ensures that the `stream` passed in from the QPS JSON driver is closed, so that the driver will terminate if the client fails to connect. If the `TryCancel` isn't there, the driver will hang forever waiting for the client to write the initial status.